### PR TITLE
Retrieve individual sites when site collections are configured

### PR DESCRIPTION
**Pull request by wangch079**
## Closes https://github.com/elastic/enterprise-search-team/issues/5739

When Site Collections are configured, SPO connector will try to fetch the site by site name one by one.

#### Pre-Review Checklist
- [x] this PR has a meaningful title
- [x] this PR links to all relevant github issues that it fixes or partially addresses
- [x] this PR has a thorough description
- [x] Covered the changes with automated tests
- [x] Tested the changes locally
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)

### CODE REVIEW

File: connectors/sources/sharepoint_online.py [hunk 1]
```diff
@@ -299,7 +299,7 @@ async def wrapped(*args, **kwargs):
                     async for item in func(*args, **kwargs, retry_count=retry):
                         yield item
                     break
-                except (NotFound, BadRequestError):
+                except (NotFound, BadRequestError, PermissionsMissing):
```
Comment (vidok) [inline connectors/sources/sharepoint_online.py:302]: It seems like adding `PermissionsMissing` is not related to the current PR. Could you please expand on this?

**Comment by vidok** [comment]
Should this also need to be backported to pre-8.10?

File: connectors/sources/sharepoint_online.py (hunk 1 above)
Comment (wangch079) [reply connectors/sources/sharepoint_online.py:302]: Correct, not related. Just feel we shouldn't retry if there's a PermissionsMissing error.

**Comment by wangch079** [comment]
>Should this also need to be backported to pre-8.10?

I think 8.10 is already released?

File: connectors/sources/sharepoint_online.py [hunk 2]
```diff
@@ -618,22 +618,24 @@ async def site_users(self, site_web_url):
...
-                # Filter out site collections that are not needed
-                if (
-                    WILDCARD not in allowed_root_sites
-                    and site["name"] not in allowed_root_sites
-                ):
-                    continue
-
                 yield site
 
+    async def site_by_site_name(self, parent_site_id, site_name):
```
Comment (seanstory) [inline connectors/sources/sharepoint_online.py:628]: This is specifically not happening by "name" but by "relative path".

**Comment by seanstory** [comment]
There's definitely overlap between https://github.com/elastic/connectors-python/pull/1653 and this, and that's on me, for not realizing that we were working on related stuff earlier. I started on mine during spacetime last week, and had intentionally not made noise about it then, but I should have been vocal come Monday. :(  Sorry!

Preferences on how to resolve the discrepancies?

**Comment by wangch079** [comment]
Close this PR as this is a duplicate of https://github.com/elastic/connectors-python/pull/1653

**Pull request closed without merging**